### PR TITLE
Adds the networkProtection remote subfeatures

### PR DESCRIPTION
**Pull request by diegoreymendez**
Task/Issue URL: https://app.asana.com/0/0/1205381133187308/f

iOS PR: https://github.com/duckduckgo/iOS/pull/1967
macOS PR: https://github.com/duckduckgo/macos-browser/pull/1562
What kind of version bump will this require?: Minor

## Description

Adds the waitlist remote flags.

## Tests:

Test using the macOS PR.

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: Sources/BrowserServicesKit/PrivacyConfig/Features/PrivacyFeature.swift [hunk 1]
```diff
@@ -64,6 +65,15 @@ public enum AutofillSubfeature: String, PrivacySubfeature {
     case autofillPasswordGeneration
 }
 
+public enum NetworkProtectionSubfeature: String, Equatable, PrivacySubfeature {
+    public var parent: PrivacyFeature {
+        .networkProtection
+    }
+
+    case waitlist
```
Comment (diegoreymendez) [inline Sources/BrowserServicesKit/PrivacyConfig/Features/PrivacyFeature.swift:73]: I'm also adding the waitlist subfeature so we can save from having to do an extra PR.

Comment (samsymons) [reply Sources/BrowserServicesKit/PrivacyConfig/Features/PrivacyFeature.swift:73]: Nice 👏

File: Sources/BrowserServicesKit/PrivacyConfig/AppPrivacyConfiguration.swift [hunk 2]
```diff
@@ -161,6 +161,11 @@ public struct AppPrivacyConfiguration: PrivacyConfiguration {
     public func isSubfeatureEnabled(_ subfeature: any PrivacySubfeature,
                                     versionProvider: AppVersionProvider,
                                     randomizer: (Range<Double>) -> Double) -> Bool {
+        // TODO: Do not ship this - temporary, only for testing
+        if let networkProtectionSubfeature = subfeature as? NetworkProtectionSubfeature, networkProtectionSubfeature == NetworkProtectionSubfeature.waitlist {
+            return true
```
Comment (samsymons) [inline Sources/BrowserServicesKit/PrivacyConfig/AppPrivacyConfiguration.swift:166]: We could remove this, and I can add checks into my waitlist branch to force the feature on where appropriate. That way we can merge this without affecting anything.

**Review by samsymons — APPROVED** [review]
LGTM!

**Pull request merged**